src/require-login.tsx

```diff
@@ -0,0 +1,171 @@
+// tslint:disable no-console
+import * as React from 'react';
+import { getKettingContext } from './provider';
+import { Client, Resource, oauth2 } from 'ketting';
+import { useHistory } from 'react-router-dom';
+
+type Props = {
+  client?: Client;
+  testEndpoint?: string | Resource;
+  clientId: string;
+  tokenEndpoint: string;
+  authorizeEndpoint: string;
+  loading: React.ReactNode;
```

Comment:
is `loading` going to be a react element?

Reply:
I haven't full figured it out yet, but I wanted allow a user to pass a component for when login is still loading. In this case i did it as a parameter, because we're already using props.children from something else... open to suggestions though.